### DISCUSSION TITLE
Dashboard Agenda Span

### DISCUSSION THREAD

**Discussion by gs-101** [emacs-dashboard/dashboard#549] [Q&A]
![agenda-span](https://github.com/user-attachments/assets/899f7234-ab83-4611-8b22-dda7001484e1)

How do I lower the agenda's span? I prefer a single-day view for my agendas, as I tend to set-up habits and daily tasks so a weekly view just ends up filled with repeated tasks. I already set `dashboard-week-agenda` to `nil`, but today is 2024-11-26 and I'm seeing tasks for tomorrow, 2024-11-27.

**Comment by ricardoricho** [comment]
Hi, this time "agenda for today" means entries with time between now and [the next 86400](https://github.com/emacs-dashboard/emacs-dashboard/blob/9479466d39652a8e32b74ee83822421fa08024d9/dashboard-widgets.el#L1455)  seconds (24 hrs).

You could define a new filter function like this:
```emacs-lisp
(defun new-agenda-filter-function ()
  "Include entries with timestamps before the end of the day."
  (let ((entry-timestamp (dashboard-agenda--entry-timestamp (point)))
        (end-of-day (days-to-time (1+ (time-to-days (current-time))))))
    (unless (and entry-timestamp ;; ensure entry has timestamp
                 (time-less-p (current-time) entry-timestamp) ;; don't show past entries
                 (time-less-p entry-timestamp end-of-day))
      (point))))
```
Or similar to [`dashboard-filter-agenda-by-time`](https://github.com/emacs-dashboard/emacs-dashboard/blob/9479466d39652a8e32b74ee83822421fa08024d9/dashboard-widgets.el#L1461), and then set `dashboard-filter-agenda-entry` to that function:
```emacs-lisp
(setq dashboard-filter-agenda-entry 'new-agenda-filter-function)
```
Hope it works, I'm not sure about the `end-of-day` value.

**Reply by gs-101** [reply]
![no-items](https://github.com/user-attachments/assets/292b1148-0596-402b-87d5-b33a15471a9f)

Thank's, but it didn't work, I set it up like this:

<details>

```emacs-lisp
(use-package dashboard-widgets
  :config
  (defun ricardorich/dashboard-day-agenda ()
    "Include entries with timestamps before the end of the day."
    (let ((entry-timestamp (dashboard-agenda--entry-timestamp (point)))
          (end-of-day (days-to-time (1+ (time-to-days (current-time))))))
      (unless (and entry-timestamp ;; Ensure that the entry has a timestamp
                   (time-less-p (current-time) entry-timestamp) ;; Don't show past entries
                   (time-less-p entry-timestamp end-of-day))
        (point))))
  :custom
  (dashboard-filter-agenda-entry 'ricardorich/dashboard-day-agenda))
```
</details>

I also evaluated the function individually with `eval-last-sexp` and used `eval-expression` for `setq dashboard-filter-agenda-entry` for good measure. But I think you may be right about `end-of-day` (Org Mode has the function `org-today`, maybe I could try that).

Besides that, I also tried setting a different name for the function, and opening regular Emacs (no daemon) as a sort of restart.
